geode-core/src/main/java/org/apache/geode/internal/cache/versions/DiskVersionTag.java

```diff
@@ -44,7 +44,7 @@ public void setMemberID(DiskStoreID memberID) {
   @Override
   public void replaceNullIDs(VersionSource memberID) {
     if (getMemberID() == null) {
-      throw new AssertionError("Member id should not be null for persistent version tags");
+      throw new IllegalStateException("Member id should not be null for persistent version tags");
```

Comment:
But we know them to be possibly `null` due to race/feature, so perhaps the oft overloaded `IllegalStateException` is too board. I would suggest an exception specific to this issue, like `NullMemberIdException`.

Reply:
Would `NullMemberIdException` extend `IllegalStateException`? We need to be concerned with how the call-path handles any runtime exceptions that are thrown. It's easy to ensure that internal-only checked exceptions don't make it to the User but runtime exceptions can easily slip through. Would you make it a new external visible User exception?

Reply:
I would rather not to introduce a new Exception class. We can use DiskAccessException here.

Reply:
`DiskAccessException` says "Indicates that an IOException during a disk region operation." so that's probably not a good fit here.

Reply:
Whatever the exception is it should be a checked exception so it has to be handled at the immediate call site and never has a chance to leak out.

Reply:
Conservation of well defined strongly named `Exception` classes is an anti-pattern. It cost us nothing to create new exceptions but cost us dearly if we don't. See [GEODE-10066](https://issues.apache.org/jira/browse/GEODE-10066) for a really good example of how throwing the inappropriate unchecked exception can cause all sorts of ugly.

Reply:
Given that throwing a checked exception from `replaceNullIDs()` resulted in pretty big cascading changes, I modified the approach to instead be to not throw an exception at all if the memberID is null, but instead check if it's still null after calling `replaceNullIDs()` and returning false from `processChunk()` if it is.

Given that we've never seen the old `AssertionError` show up anywhere else, and that it wasn't being handled anywhere, removing the exception throwing behaviour entirely should be safe.